src/main/java/io/odpf/firehose/factory/FirehoseConsumerFactory.java

```diff
@@ -87,7 +90,8 @@ public FirehoseConsumerFactory(KafkaConsumerConfig kafkaConsumerConfig, StatsDRe
      */
     public FirehoseConsumer buildConsumer() {
 
-        Filter filter = new MessageFilter(kafkaConsumerConfig, new Instrumentation(statsDReporter, MessageFilter.class));
+        Filter filter = (kafkaConsumerConfig.getFilterEngine() == JEXL) ? new JexlFilter(kafkaConsumerConfig, new Instrumentation(statsDReporter, JexlFilter.class)) : new JsonFilter(kafkaConsumerConfig, new Instrumentation(statsDReporter, JsonFilter.class));
```

Comment:
why not have a FilterConfig instead of passing kafkaconsumerconfig? Most of the configuration have FILTER_  prefix.

Reply:
fixed